src/lib.rs

```diff
@@ -2,269 +2,209 @@
 //
 // SPDX-License-Identifier: MIT OR Apache-2.0
 
-/*! The library provide tools for handling compressed and archive files
-
-# Examples
-```no_run
-use compress_tools;
-
-let dir = tempfile::tempdir().unwrap();
-compress_tools::uncompress("tests/fixtures/tree.tar.gz", dir.path(), compress_tools::Kind::TarGZip).unwrap();
-```
-*/
-
-mod uncompress;
-
-pub use uncompress::{Error, Result};
-
-use std::path::Path;
-
-/// Type of compressed file or archive
-#[derive(Copy, Clone, Debug)]
-pub enum Kind {
-    TarGZip,
-    TarBZip2,
-    TarXz,
-    TarLZMA,
-    TarLZip,
-    Tar,
-    Zip,
-
-    GZip,
-    BZip2,
-    Xz,
-    LZip,
-    LZMA,
+mod error;
+mod ffi;
+
+pub use error::*;
+use ffi::*;
```

Comment:
It's better if we don't use glob imports here too, se the code always shows by itselfs where everything comes from.